internal/core/writer/writer.go

```diff
@@ -115,6 +115,10 @@ func New(conf *config.WriterConfig, dpChan chan []*datapoint.Datapoint, eventCha
 		dpChan:            dpChan,
 		spanChan:          spanChan,
 	}
+	if conf.ExportSAPM() {
+		sw.client = sfxclient.WithSAPMExporter(sw.client)
+	}
```

Comment:
Nit: move `sfxclient.NewHTTPSink()` case to `else` here for symmetry?

Reply:
Yes, it would be a bit clearer to see the two distinct clients set closer together in code.

I'm assuming the WithSAPMExporter client behaves the same as the NewHTTPSink client for datapoints and events?

Reply:
Updated by using the options pattern

Reply:
> I'm assuming the WithSAPMExporter client behaves the same as the NewHTTPSink client for datapoints and events?

Yes, it only affect trace exporter. I'll update it to be more descriptive.

Reply:
Updated